odl/contrib/fom/supervised/supervised_fom.py

```diff
@@ -0,0 +1,575 @@
+    else:
+        data_range = np.max(data) - np.min(data)
+        ground_truth_range = np.max(ground_truth) - np.min(ground_truth)
+
+    fom = np.abs(data_range - ground_truth_range)
+
+    if normalized:
+        fom /= np.abs(data_range + ground_truth_range)
+
+    return fom
+
+
+def blurring(data, ground_truth, mask=None, normalized=False,
+             weight_factor=30):
+    """FOM returning weighted L2-distance between ``data`` and ``ground_truth``,
+        emphasizing regions defined by ``mask``.
+
+    Evaluates `mean squared error
+    <https://en.wikipedia.org/wiki/Mean_squared_error>`_ between input
+    (``data``) and reference data (``ground_truth``) using an added binary
+    mask (``mask``), such that the error is weighted with higher importance
+    given to the defined structure-of-interest. Allows for normalization
+    (``normalized``).
+
+    NOTE: If omitting the mask argument, the blurring FOM is equivalent to the
+    mean squared error FOM.
+
+    Notes
+    ----------
+    The FOM evaluates
+
+    .. math::
+        \|\\alpha (f - g) \|^2_2,
+
+    or, in normalized form
+
+    .. math::
+        \\frac{\| \\alpha (f - g) \|^2_2}{\| \\alpha f \|^2_2 +
+                                  \| \\alpha  g \|^2_2},
+
+    where :math:`\\alpha` is a weighting function with higher values near a
+    structure of interest defined by ``mask``. The weighting function is given
+    as
+
+    .. math::
+        \\alpha = e^{-\\frac{1}{k} \\beta},
+
+    where :math:`\\beta(x)` is the Euclidian distance transform from :math:`x`
+    to the complement of the structure of interest, and :math:`k` is a positive
+    real number.
+
+    The normalized FOM takes its values in [0, 1], with higher correspondance
+    at lower FOM value.
+
+    Parameters
+    ----------
+    data : `FnBaseVector`
+        Input data or reconstruction.
+    ground_truth : `FnBaseVector`
+        Reference to compare ``data`` to.
+    mask : `FnBaseVector`
+        Binary mask to define ROI in which FOM evaluation is performed.
+    normalized  : bool
+        Boolean flag to switch between unormalized and normalized FOM.
+    weight_factor : float
+        Positive real number for scaling of weighted mask. Higher value gives
+        smoother weighting.
+
+    Returns
+    -------
+    fom : float
+        Scalar (float) indicating weighted mean squared error between
+        ``data`` and ``ground_truth``. In normalized form the FOM takes
+        values in [0, 1], with higher correspondance at lower FOM value.
+    """
+
+    import numpy as np
+    import odl.contrib.fom
+    import scipy.ndimage.morphology as scimorph
+
+    if mask is not None:
+        mask = scimorph.distance_transform_edt(1 - mask)
+        mask = np.exp(-mask / weight_factor)
+
+    fom = odl.contrib.fom.mean_square_error(data,
+                                            ground_truth,
+                                            mask=mask,
```

Comment:
Not at all obvious that this is legal given that you say in the `mean_square_error` doc that `mask` is binary. I guess that doc has to change.